src/repository.rs

```diff
@@ -289,6 +290,8 @@ where
     }
 }
 
+/// A custom hook to be invoked before sending HTTP requests.
+pub type HttpHook = Box<for<'a> Fn(RequestBuilder<'a>) -> RequestBuilder<'a>>;
```

Comment:
Ok, so I did my research and revitalized I misunderstood `catch_unwind`. Don't wrap call site, actually. Turns out that's a bad idea.